packages/teleport-types/src/uidl.ts

```diff
@@ -318,36 +299,50 @@ export interface UIDLCMSItemNode {
   content: UIDLCMSItemNodeContent
 }
 
+/*
+  A cms-list node can fetch data from the remote resouce
+  or it can refer to a `prop` value for page list.
+  It can have either remote resource or prop but not both.
+  In the case of `prop` the data is fetched using `getStaticProps`
+  or `getStaticPaths` at the moment in NextJS.
+*/
+
+export interface UIDLResourceLink {
+  id: string
+  params?: Record<string, UIDLStaticValue | UIDLPropValue | UIDLExpressionValue>
+}
+
 export interface UIDLCMSListNodeContent {
+  name: string
+  key: string // internal usage
+  attrs?: Record<string, UIDLAttributeValue>
   nodes: {
     success: UIDLElementNode
     error?: UIDLElementNode
     loading?: UIDLElementNode
     empty?: UIDLElementNode
   }
-  statePersistanceName?: string
-  loopItemsReference?: UIDLAttributeValue
+  cmsIdentifier: string
   valuePath?: string[]
   itemValuePath?: string[]
-  resource: {
-    id: string
-    params?: Record<string, UIDLStaticValue | UIDLPropValue | UIDLExpressionValue>
-  }
+  resource?: UIDLResourceLink
+  initialData?: UIDLPropValue
 }
 
 export interface UIDLCMSItemNodeContent {
+  name: string
+  key: string // internal usage
+  attrs?: Record<string, UIDLAttributeValue>
+  cmsIdentifier: string
   nodes: {
     success: UIDLElementNode
     error?: UIDLElementNode
     loading?: UIDLElementNode
   }
-  statePersistanceName?: string
   valuePath?: string[]
   itemValuePath?: string[]
```

Comment:
Right now, it takes a lot of changes. But down the line we should merge this. What say @ionutpasca. It was little confusing me till now, to have them separately. Because the node types are separate. And one API call can have only one of them and not both. So we can just keep `path` alone instead of two variables.